cart-service/src/tests/service-endpoint.tests.ts

```diff
@@ -244,6 +244,125 @@ test('should return action to set coupon codes on cart when Dovetech service ret
   });
 });
 
+test('should return set commitId action when type is Order', async () => {
+  const currencyCode = 'EUR';
+  const amountOff = 10;
+  const total = 22.97;
+
+  const ctCart = orderWithEvaluationResult as CartOrOrder;
+
+  const amountOffBasketAction: AmountOffAction = buildAmountOffAction(
+    DoveTechActionType.AmountOffBasket,
+    amountOff
+  );
+
+  const dtResponse = new DoveTechResponseBuilder()
+    .addCommitId('123')
+    .addAction(amountOffBasketAction)
+    .addLineItem({
+      totalAmountOff: amountOffBasketAction.amountOff,
+      total: total,
+      actions: [
+        {
+          id: amountOffBasketAction.id,
+          subItemId: 0,
+          amountOff: amountOffBasketAction.amountOff,
+        },
+      ],
+    })
+    .build();
+
+  fetchMock.mockResponseOnce(JSON.stringify(dtResponse));
+
+  const response = await postCart(ctCart);
+
+  expect(response.status).toBe(200);
+
+  expect(response.body).toEqual({
+    actions: [
+      buildSetCustomTypeAction(  dtResponse, currencyCode, '[]' ),
+    ],
+  });
+});
+
+test('should return no discount actions when type is Order', async () => {
+  const currencyCode = 'EUR';
+  const amountOff = 10;
+  const total = 22.97;
+
+  const ctCart = orderWithEvaluationResult as CartOrOrder;
+
+  const amountOffBasketAction: AmountOffAction = buildAmountOffAction(
+    DoveTechActionType.AmountOffBasket,
+    amountOff
+  );
+
+  const dtResponse = new DoveTechResponseBuilder()
+    .addCommitId('123')
+    .addAction(amountOffBasketAction)
+    .addLineItem({
+      totalAmountOff: amountOffBasketAction.amountOff,
+      total: total,
+      actions: [
+        {
+          id: amountOffBasketAction.id,
+          subItemId: 0,
+          amountOff: amountOffBasketAction.amountOff,
+        },
+      ],
+    })
+    .build();
+
+  fetchMock.mockResponseOnce(JSON.stringify(dtResponse));
+
+  const response = await postCart(ctCart);
+
+  expect(response.status).toBe(200);
+
+  expect(response.body).toEqual({
+    actions: [buildSetCustomTypeAction(dtResponse, currencyCode, '[]')],
+  });
+});
+
+test('should return no coupocode accepted actions when type is Order', async () => {
```

Comment:
I'm not really sure what this is testing? The order doesn't have any coupon codes on it, so the logic seems to be covered by the "should return set commitId action when type is Order" test?

Reply:
removed test